contracts/MigrationContractStaging.cdc

```diff
@@ -0,0 +1,362 @@
+///
+/// This can be done in a single transaction! For more code context, see https://github.com/onflow/contract-updater
+///
+access(all) contract MigrationContractStaging {
+
+    // Path constants
+    //
+    access(self) let delimiter: String
+    access(self) let capsulePathPrefix: String
+    access(all) let HostStoragePath: StoragePath
+    access(all) let AdminStoragePath: StoragePath
+    /// Maps contract addresses to an array of staged contract names
+    access(self) let stagedContracts: {Address: [String]}
+    /// The block height at which updates can no no longer be staged. If nil, updates can be staged indefinitely until
+    /// the cutoff value is set.
+    access(all) var stagingCutoff: UInt64?
+
+    /// Event emitted when a contract's code is staged
+    /// status == true - insert | staged == false - replace | staged == nil - remove
+    /// NOTE: Does not guarantee that the contract code is valid Cadence
+    access(all) event StagingStatusUpdated(
+        capsuleUUID: UInt64,
+        address: Address,
+        codeHash: [UInt8],
+        contract: String,
+        status: Bool?
+    )
+    /// Emitted when the stagingCutoff value is updated
+    access(all) event StagingCutoffUpdated(old: UInt64?, new: UInt64?)
+
+    /********************
+        Public Methods
+     ********************/
+
+    /* --- Staging Methods --- */
+
+    /// 1 - Create a host and save it in your contract-hosting account at MigrationContractStaging.HostStoragePath
+    ///
+    /// Creates a Host serving as identification for the contract account. Reference to this resource identifies the
+    /// calling address so it must be saved in account storage before being used to stage a contract update.
+    ///
+    access(all) fun createHost(): @Host {
+        return <- create Host()
+    }
+
+    /// 2 - Call stageContract() with the host reference and contract name and contract code you wish to stage
+    ///
+    /// Stages the contract code for the given contract name at the host address. If the contract is already staged,
+    /// the code will be replaced.
+    ///
+    access(all) fun stageContract(host: &Host, name: String, code: String) {
+        pre {
+            self.stagingCutoff == nil || self.stagingCutoff! > getCurrentBlock().height: "Staging period has ended"
+        }
+        let capsulePath: StoragePath = self.deriveCapsuleStoragePath(contractAddress: host.address(), contractName: name)
+        if self.stagedContracts[host.address()] == nil {
+            // First time we're seeing contracts from this address - insert the address and contract name
+            self.stagedContracts.insert(key: host.address(), [name])
+            // Create a new Capsule to store the staged code
+            let capsule: @Capsule <- self.createCapsule(host: host, name: name, code: code)
+            self.account.save(<-capsule, to: capsulePath)
```

Comment:
nit: i prefer return home early here. if you just return after this you dont need the else, and the code becomes IMHO easier to read. Now you have to check below the else to see if there are more going on on this code path.

Reply:
Makes sense, updated!